File: app/src/main/java/net/tlalka/android/fiszki/view/navigations/Navigator.java
package net.tlalka.android.fiszki.view.navigations;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.os.Parcelable;
import net.tlalka.android.fiszki.model.dto.LessonDto;
import net.tlalka.android.fiszki.model.dto.WelcomeDto;
import net.tlalka.android.fiszki.view.activities.LessonActivity;
import net.tlalka.android.fiszki.view.activities.LessonListActivity;
import net.tlalka.android.fiszki.view.activities.LessonScoreActivity;
import net.tlalka.android.fiszki.view.activities.TestActivity;
import net.tlalka.android.fiszki.view.activities.TestListActivity;
import net.tlalka.android.fiszki.view.activities.TestScoreActivity;
import net.tlalka.android.fiszki.view.activities.WelcomeActivity;
import org.parceler.Parcels;

import javax.inject.Inject;
import javax.inject.Singleton;


@Singleton
public class Navigator {

    @Inject
    public Navigator() {
    }

    public void openWelcomeActivity(Context context, WelcomeDto welcomeDto) {
        Parcelable parcelable = Parcels.wrap(welcomeDto);

        Intent intent = new Intent(context, WelcomeActivity.class);
        intent.putExtra(WelcomeDto.class.getName(), parcelable);
        context.startActivity(intent);
    }

    public void openLessonActivity(Context context, LessonDto lessonDto) {
        Parcelable parcelable = Parcels.wrap(lessonDto);

        Intent intent = new Intent(context, LessonActivity.class);
        intent.putExtra(LessonDto.class.getName(), parcelable);
        context.startActivity(intent);
    }

    public void openLessonListActivity(Context context) {
        context.startActivity(new Intent(context, LessonListActivity.class));
    }

    public void openLessonScoreActivity(Context context, LessonDto lessonDto) {
        Parcelable parcelable = Parcels.wrap(lessonDto);

        Intent intent = new Intent(context, LessonScoreActivity.class);
        intent.putExtra(LessonDto.class.getName(), parcelable);
        context.startActivity(intent);
    }

    public void openTestActivity(Context context, LessonDto lessonDto) {
        Parcelable parcelable = Parcels.wrap(lessonDto);

        Intent intent = new Intent(context, TestActivity.class);
        intent.putExtra(LessonDto.class.getName(), parcelable);
        context.startActivity(intent);
    }

    public void openTestListActivity(Context context) {
        context.startActivity(new Intent(context, TestListActivity.class));
    }

    public void openTestScoreActivity(Context context, LessonDto lessonDto) {
        Parcelable parcelable = Parcels.wrap(lessonDto);

        Intent intent = new Intent(context, TestScoreActivity.class);
        intent.putExtra(LessonDto.class.getName(), parcelable);
        context.startActivity(intent);
    }

    public void finish(Activity activity) {
        activity.finish();
    }
}

File: app/src/main/kotlin/net/tlalka/fiszki/view/navigations/Navigator.kt
package net.tlalka.fiszki.view.navigations

import android.app.Activity
import android.content.Context
import android.content.Intent
import net.tlalka.fiszki.model.dto.LessonDto
import net.tlalka.fiszki.model.dto.WelcomeDto
import net.tlalka.fiszki.view.activities.*
import org.parceler.Parcels
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class Navigator @Inject constructor() {

    fun openWelcomeActivity(context: Context, welcomeDto: WelcomeDto) {
        val intent = Intent(context, WelcomeActivity::class.java)
        intent.putExtra(WelcomeDto::class.java.name, Parcels.wrap(welcomeDto))
        context.startActivity(intent)
    }

    fun openLessonActivity(context: Context, lessonDto: LessonDto) {
        val intent = Intent(context, LessonActivity::class.java)
        intent.putExtra(LessonDto::class.java.name, Parcels.wrap(lessonDto))
        context.startActivity(intent)
    }

    fun openLessonListActivity(context: Context) {
        context.startActivity(Intent(context, LessonListActivity::class.java))
    }

    fun openLessonScoreActivity(context: Context, lessonDto: LessonDto) {
        val intent = Intent(context, LessonScoreActivity::class.java)
        intent.putExtra(LessonDto::class.java.name, Parcels.wrap(lessonDto))
        context.startActivity(intent)
    }

    fun openTestActivity(context: Context, lessonDto: LessonDto) {
        val intent = Intent(context, TestActivity::class.java)
        intent.putExtra(LessonDto::class.java.name, Parcels.wrap(lessonDto))
        context.startActivity(intent)
    }

    fun openTestListActivity(context: Context) {
        context.startActivity(Intent(context, TestListActivity::class.java))
    }

    fun openTestScoreActivity(context: Context, lessonDto: LessonDto) {
        val intent = Intent(context, TestScoreActivity::class.java)
        intent.putExtra(LessonDto::class.java.name, Parcels.wrap(lessonDto))
        context.startActivity(intent)
    }

    fun openWordsActivity(context: Context, lessonDto: LessonDto) {
        val intent = Intent(context, WordsActivity::class.java)
        intent.putExtra(LessonDto::class.java.name, Parcels.wrap(lessonDto))
        context.startActivity(intent)
    }

    fun finish(activity: Activity) {
        activity.finish()
    }
}
